### PR TITLE
ADAPT-2706 ADAPT-2113: Masthead Styles

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Desktop masthead styles
- Mobile masthead styles (exclude the actual mobile main nav and the toggle behavior of the menu button)
- Button that opens up the search overlay without the functionality (just styling)
- Fixup Basic card width (fit 100% to parent container, this matters for cards with really short content)

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. https://deploy-preview-39--stanford-alumni.netlify.app/test/yvonne/test-basic-page-interior-hero/
2. https://deploy-preview-39--stanford-alumni.netlify.app/test/yvonne/test-basic-page-interior-no-hero/
3. Look at both of the above previews, one with a hero banner layered beneath, the other without.
4. Make sure to check all breakpoints carefully, please expand your browser width slowly.
5. Desktop breakpoints of particular interest - 992 (LG), 1200 (XL), 1500 (2XL), 1700 (3XL = 2XL + 2 times the screen margin of 100px on each side).

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: src/components/search/openSearchModalButton.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { Button } from "decanter-react";
+import { SearchIcon } from "@heroicons/react/solid";
+
+const OpenSearchModalButton = (props) => (
```
Comment (yvonnetangsu) [inline src/components/search/openSearchModalButton.js:5]: @nplowman @romakravchuk FYI I've started styling the button in the masthead that opens up the search modal, but I won't be attaching any functionalities to it.

**Comment by yvonnetangsu** [comment]
@sherakama This is ready for your review - no need to look before sprint close or anything 😄 

I did, sadly, discover an issue with using the Popover for the main nav - I found that on Safari and Firefox, once a dropdown is open, I need to click twice to close it. On Chrome, it's working as intended.

I've tried a few things, but haven't found a solution yet 🤔  However, if possible, I'd like to get this PR merged in first to get the masthead styles in and I'll create a ticket for this bug.

**Comment by sherakama** [comment]
Screen Shot 2021-06-17 at 1 49 36 PM

Probably not worth addressing until we get feedback from SODA but at 200% zoom on mobile things get a little squishy. Just an FYI.

File: src/components/navigation/mainMenuGroup.js [hunk 2]
```diff
@@ -23,7 +23,7 @@ const MainMenuGroup = ({
           {parentTextSecond && (
             <>
               <br className="su-hidden xl:su-inline 2xl:su-hidden" />
-              {` ${parentTextSecond}`}
+              {`${parentTextSecond}`}
```
Comment (sherakama) [inline src/components/navigation/mainMenuGroup.js:26]:
```suggestion
              {parentTextSecond}
```

Comment (sherakama) [reply src/components/navigation/mainMenuGroup.js:26]: Going ahead with this comment so we can get this merged.

**Review by sherakama — APPROVED** [review]
GTG

File: src/components/navigation/mainMenuGroup.js (hunk 2 above)
Comment (yvonnetangsu) [reply src/components/navigation/mainMenuGroup.js]: Thank you!

**Comment by yvonnetangsu** [comment]
> Screen Shot 2021-06-17 at 1 49 36 PM
> 
> Probably not worth addressing until we get feedback from SODA but at 200% zoom on mobile things get a little squishy. Just an FYI.

Thanks for calling that out! Well....Kevin did say things doesn't need to look good just functional, so we'll see 😂

**Pull request merged**